src/main/java/net/server/channel/handlers/RemoteStoreHandler.java

```diff
@@ -39,6 +41,13 @@ public void handlePacket(InPacket p, Client c) {
         HiredMerchant hm = getMerchant(c);
         if (hm != null && hm.isOwner(chr)) {
             if (hm.getChannel() == chr.getClient().getChannel()) {
+                boolean isPlayerOnMerchantMap = chr.getMapId() == hm.getMapId();
+                if (!isPlayerOnMerchantMap) {
+                    var remoteControl = chr.getInventory(InventoryType.CASH).findById(ItemId.REMOTE_CONTROLLER);
+                    if (remoteControl == null) {
+                        return;
+                    }
```

Comment:
Good check, but it would be better placed at the start of the method. Now the check is not made if the player is on a different channel.